lib/ansible/executor/task_executor.py

```diff
@@ -448,6 +448,18 @@ def _execute(self, templar: TemplateEngine, variables: dict[str, t.Any]) -> dict
 
         return result
 
+    def _update_task_connection(self, task: Task | None = None) -> None:
+        """If a connection plugin is loaded, ensure the resolved name is propagated back to the controller as the task's connection."""
+
+        if not task:
+            task = self._task
+
+        # FUTURE: What value should be reported when there is no connection?
+        #         This is currently not possible, but it should be.
+
+        if isinstance(self._connection, ConnectionBase):
```

Comment:
When would this ever not be `ConnectionBase`? Just a guard in case a plugin changes the `_connection` attribute to something else?